Starwatch.Core/API/Rest/Route/AnnouncementRoute.cs

```diff
@@ -0,0 +1,387 @@
+        const string MISSING_RESOURCE = "No such announcement.";
+
+        const string ID_MUST_BE_INTEGER = "id must be an integer.";
+        const string ID_MUST_BE_POSITIVE = "id must be greater than or equal to 0.";
+
+        public AnnouncementRoute(RestHandler handler, Authentication authentication) : base(handler, authentication) { }
+        public override Type PayloadType => typeof(AnnouncementPatch);
+        public string GetRouteName() => "/announcement";
+
+        public void SetGateway(EventConnection gateway)
+        {
+            Handler = gateway.API.RestHandler;
+            Authentication = gateway.Authentication;
+        }
+
+        /// <summary>
+        /// Gets the account exists and its admin state.
+        /// </summary>
+        /// <param name="query"></param>
+        /// <returns></returns>
+        public override RestResponse OnGet(Query query)
+        {
+#if DEBUG
+            if (Debugger.IsAttached)
+            {
+                if (query.ContainsKey("dbg"))
+                {
+                    if (query["dbg"] == "post")
+                    {
+                        return OnPost(query, null);
+                    }
+
+                    if (query["dbg"] == "delete")
+                    {
+                        return OnDelete(query);
+                    }
+
+                    if (query["dbg"] == "patch")
+                    {
+                        return OnPatch(query, null);
+                    }
+                }
+            }
+#endif
+            // Grab the configurator from the monitor 
+            AnnouncementMonitor mon = null;
+            bool hasMon = Starbound.TryGetMonitor(out mon);
+
+            if (!hasMon)
+            {
+                return new RestResponse(RestStatus.InternalError, "AnnouncementMonitor was not found.");
+            }
+
+            AnnouncementMonitor.Announcement[] announcements = mon.Announcements;
+
+            if (query.ContainsKey("id"))
+            {
+                // Grabs a specific announcement.
+                int id = 0;
+                bool ok = int.TryParse(query["id"], out id);
+                if (!ok)
+                {
+                    // probably unnecessary
+                    id = 0;
+                }
+
+                if (id < 0)
+                {
+                    return new RestResponse(RestStatus.BadRequest, ID_MUST_BE_POSITIVE);
+                }
+
+                if (announcements.Length > id)
+                {
+                    return new RestResponse(RestStatus.OK, announcements[id]);
+                }
+
+                return new RestResponse(RestStatus.ResourceNotFound, MISSING_RESOURCE);
+            }
+            else
+            {
+                // Returns all announcements
+                if (announcements.Length == 1)
+                {
+                    return new RestResponse(RestStatus.OK, "There is 1 announcement.", 1);
+                }
```

Comment:
Messages are kinda just to tell the consumer what's crackalakin. So if there is an error or some kind of status to report on. This is fine, but dont go into extra lengths to make it super human readable 👍

Reply:
Yeah it probably is a waste of resources in the long-run 👍